### DISCUSSION TITLE
Make Styles apply to Elements

### DISCUSSION THREAD

**Discussion by mike-rowley** [dotnet/maui#15284] [Ideas]
In trying to apply a Style to a Popup using this code in Styles.xaml
```
    <Style TargetType="toolkit:Popup">
        <Setter Property="Color" Value="{AppThemeBinding Light={StaticResource White}, Dark={StaticResource Black}}" />
    </Style>
```
It was discovered that Popup is an Element and not a VisualElement because it can't be a ViewHandler the reason is to be a ViewHandler the control should inherit from what represents a View on each platform, (UIView on apple, View on Android, etc) the popup doesn't have that so it should be an ElementHandler and because of that an Element on the Maui world

I believe not allowing Styles on Elements is a huge oversight and creates an inconsistency for developers. Creating a situation where we have to know the base classes of anything to know if Styles works or not, that doesn't seem to make any sense and creates a situation where Styles works for some controls and not others, that inconsistency is one of the many failings of Xamarin that lead to the creation of MAUI, hopefully we can correct these inconsistencies.
I'm sure there are technical considerations at play as to why it was done this way but IMO those are far less important than presenting a complete, consistent and usable Styling interface for developers.

In talking with the folks in the CommunityToolkit repo they suggested I elevate my question there to the main Maui repo, here is the [original conversation for reference](https://github.com/MicrosoftDocs/CommunityToolkit/issues/277)

**Comment by Xyncgas** [comment]
why not use css frameworks like boostrap it is so simple to do
